### PR TITLE
DEV-307 Adapt middleware deployment script for testnet

### PR DESCRIPTION
**Pull request by yonikesel1**
1. Restructure deployment folders
2. Added deployment config
3. Added sanity checks after deployment

### CODE REVIEW

File: script/deploy/goerli/Goerli_DeployEOMiddlewareContracts.s.sol [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+    EOIndexRegistry indexRegistry;
+    OperatorStateRetriever operatorStateRetriever;
+
+    EORegistryCoordinator registryCoordinatorImplementation;
+    EOStakeRegistry stakeRegistryImplementation;
+    EOBLSApkRegistry blsApkRegistryImplementation;
+    EOIndexRegistry indexRegistryImplementation;
+    EOServiceManager serviceManagerImplementation;
+
+    uint256 numStrategies = deployedStrategyArray.length;
```
Comment (RonTuretzky) [inline script/deploy/goerli/Goerli_DeployEOMiddlewareContracts.s.sol]: I don't understand how this is a legal line lol , where is deployedStrategyArray defined?

File: test/integration/tests/DeployEOMiddlewareContracts.t.sol [hunk 2]
```diff
@@ -20,7 +20,9 @@ import {EOServiceManager} from "../../../src/EOServiceManager.sol";
 import {OperatorStateRetriever} from "../../../src/OperatorStateRetriever.sol";
 
 contract DeployEOMiddlewareContractsTest is Test, Script {
-    DeployEOMiddlewareContracts public deployEOMiddlewareContracts;
+    Goerli_DeployEOMiddlewareContracts public deployEOMiddlewareContracts;
+
+    address public EORACLE_OWNER = 0xeA96Fa8F6a185D6B97eCDeb9805178F6C2829eeE;
```
Comment (RonTuretzky) [inline test/integration/tests/DeployEOMiddlewareContracts.t.sol]: this should be changed to 0x119156A81E4f3d891cBd106a4AedEA916A0A9C4b as we stopped using the Ae96 account outside of local deploy

**Review by RonTuretzky — CHANGES_REQUESTED** [review]

File: test/integration/tests/DeployEOMiddlewareContracts.t.sol (hunk 2 above)
Comment (RonTuretzky) [inline test/integration/tests/DeployEOMiddlewareContracts.t.sol]:
```suggestion
    address public EORACLE_OWNER = 0x119156A81E4f3d891cBd106a4AedEA916A0A9C4b;
```

**Review by RonTuretzky — CHANGES_REQUESTED** [review]

File: script/deploy/goerli/Goerli_DeployEOMiddlewareContracts.s.sol (hunk 1 above)
Comment (yonikesel1) [reply script/deploy/goerli/Goerli_DeployEOMiddlewareContracts.s.sol]: deployedStrategyArray is defined in ExistingDeploymentParser.sol.
But, numStrategies is not used here so removing it.

File: test/integration/tests/DeployEOMiddlewareContracts.t.sol (hunk 2 above)
Comment (yonikesel1) [reply test/integration/tests/DeployEOMiddlewareContracts.t.sol]: done

Comment (yonikesel1) [reply test/integration/tests/DeployEOMiddlewareContracts.t.sol]: done

**Review by RonTuretzky — APPROVED** [review]

**Review by ArielLaub — APPROVED** [review]

**Pull request merged**